src/components/GlobalProgress.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div class="progress-display" v-if="display_progress">
+    <p class="caption">{{ message }}</p>
+    <n-progress type="line" :status="status" :percentage="percentage" indicator-placement="inside"
+      :show-indicator="false" :height="30" border-radius="2px" fill-border-radius="2px"></n-progress>
+  </div>
+</template>
+
+<script>
+import { NProgress } from 'naive-ui'
+import { defineComponent, h } from 'vue'
+import { listen } from '@tauri-apps/api/event'
+
+export default {
+  name: 'GlobalProgress',
+  components: { NProgress },
+  data: () => ({
+    percentage: "0",
+    status: "info",
+    message: "",
+    targets: [],
+    display_progress: false,
+  }),
+  methods: {
+    startListening() {
+      console.log('Listening for progress messages...');
+      listen('progress-message', (event) => {
+        // console.log('Received progress message:', event);
+        this.message = event.payload.message;
+        this.status = event.payload.status;
+        this.percentage = event.payload.percentage;
+        this.display_progress = event.payload.display;
```

Comment:
Can we capture the progress messages to a file so that later, if there are any issues, it would be helpful for troubleshooting?

Reply:
logging is configured here: https://github.com/espressif/idf-im-ui/blob/master/src-tauri/src/lib.rs#L345
right now we do not use any file, but will follow convention from eim-cli